pkg/extender/extender.go

```diff
@@ -730,6 +754,48 @@ sendResponse:
 	}
 }
 
+// isUsingBindMount returns true if the Pod is running on the same node where the volume
+// is attached
+func (e *Extender) isUsingBindMount(pod *v1.Pod, vol *volume.Info) bool {
+	log.Debugf("Pod Name: %v, Pod Node IP: %v,  Volume Node IP: %v", pod.Name, pod.Status.HostIP, vol.AttachedOn)
+	return pod.Status.HostIP == vol.AttachedOn
+}
+
+// getLiveMigrationInfo returns status of live migration
+func (e *Extender) getLiveMigrationInfo(refPod *v1.Pod, refVMIUID types.UID) (podBeingLiveMigrated *v1.Pod, err error) {
+	pods, err := core.Instance().GetPods(refPod.Namespace, nil)
+	if err != nil {
+		msg := "unable to get Pod list in the same namespace."
+		return nil, fmt.Errorf("err: %s", msg)
+	}
+	// If another Running Pod in the same namespace which also prefersBindMount and has the same VMI owner reference
+	// Then it can be concluded that this Pod is where the original VM was being hosted and is being live migrated to the refPod
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != v1.PodRunning ||
+			!e.Driver.PodPrefersBindMount(&pod) {
```

Comment:
Why is this `PodPrefersBindMount` check required here? This function is invoked only when `processVirtLauncherPod` function is executed which in itself is guarded by the if check of `PodPrefersBindMount`

Reply:
The function is invoked in context of refPod scheduling. In this loop we are iterating over all other Pods its namespace and trying to find the existing Pod which is being live migrated so we need this check.